src/django_components/component.py

```diff
@@ -1128,26 +1263,56 @@ def _normalize_slot_fills(
 
         # NOTE: `gen_escaped_content_func` is defined as a separate function, instead of being inlined within
         #       the forloop, because the value the forloop variable points to changes with each loop iteration.
-        def gen_escaped_content_func(content: SlotFunc) -> Slot:
-            def content_fn(ctx: Context, slot_data: Dict, slot_ref: SlotRef) -> SlotResult:
-                rendered = content(ctx, slot_data, slot_ref)
-                return conditional_escape(rendered) if escape_content else rendered
+        def gen_escaped_content_func(content: SlotFunc, slot_name: str) -> Slot:
```

Comment:
This is part of the `_normalize_slot_fills` method, which ensures that the outputs of slot fills are escaped if configured so.

I updated this function to set the component and slot names onto the `Slot` objects. This is what makes it possible to print a user-friend representation when  printing out the slots, e.g.

<img width="595" alt="Screenshot 2025-01-28 at 09 52 00" src="https://github.com/user-attachments/assets/b791e7e0-1800-4439-bbd5-bbde8d524d37" />